plugins/source/aws/client/table_options_test.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+)
+
+// TestTableOptionsUnmarshal tests that the TableOptions struct can be unmarshaled from JSON using
+// snake_case keys.
+func TestTableOptionsUnmarshal(t *testing.T) {
+	opts := `{
+	"aws_cloudtrail_events": {
+		"lookup_events": {
+			"start_time": "2020-01-01T00:00:00Z",
+			"end_time": "2020-01-02T00:00:00Z",
+			"event_category": "insight",
+			"lookup_attributes": [
+				{
+					"attribute_key": "EventName",
+					"attribute_value": "ConsoleLogin"
+				}
+			]
+		}
+	}
+}`
+	var tableOpts TableOptions
+	if err := json.Unmarshal([]byte(opts), &tableOpts); err != nil {
+		t.Fatal(err)
+	}
+	if tableOpts.CloudTrailEvents == nil {
+		t.Fatal("CloudTrailEvents is nil")
+	}
+	start := time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC)
+	gotStart := tableOpts.CloudTrailEvents.LookupEventsOpts.StartTime
+	if gotStart == nil || !gotStart.Equal(start) {
+		t.Fatalf("CloudTrailEvents.LookupEventsOpts.StartTime = %s, want %s", tableOpts.CloudTrailEvents.LookupEventsOpts.StartTime, start)
+	}
+	end := time.Date(2020, 1, 2, 0, 0, 0, 0, time.UTC)
+	gotEnd := tableOpts.CloudTrailEvents.LookupEventsOpts.EndTime
+	if gotEnd == nil || !gotEnd.Equal(end) {
+		t.Fatalf("CloudTrailEvents.LookupEventsOpts.EndTime = %s, want %s", tableOpts.CloudTrailEvents.LookupEventsOpts.EndTime, end)
+	}
+	if tableOpts.CloudTrailEvents.LookupEventsOpts.EventCategory != "insight" {
+		t.Fatalf("CloudTrailEvents.LookupEventsOpts.EventCategory = %s, want insight", tableOpts.CloudTrailEvents.LookupEventsOpts.EventCategory)
+	}
+	if len(tableOpts.CloudTrailEvents.LookupEventsOpts.LookupAttributes) != 1 {
+		t.Fatalf("CloudTrailEvents.LookupEventsOpts.LookupAttributes = %v, want 1 item", tableOpts.CloudTrailEvents.LookupEventsOpts.LookupAttributes)
+	}
+	if tableOpts.CloudTrailEvents.LookupEventsOpts.LookupAttributes[0].AttributeKey != "EventName" {
+		t.Fatalf("CloudTrailEvents.LookupEventsOpts.LookupAttributes[0].AttributeKey = %v, want EventName", tableOpts.CloudTrailEvents.LookupEventsOpts.LookupAttributes[0].AttributeKey)
+	}
+	if *tableOpts.CloudTrailEvents.LookupEventsOpts.LookupAttributes[0].AttributeValue != "ConsoleLogin" {
+		t.Fatalf("CloudTrailEvents.LookupEventsOpts.LookupAttributes[0].AttributeKey = %v, want EventName", *tableOpts.CloudTrailEvents.LookupEventsOpts.LookupAttributes[0].AttributeValue)
+	}
+}
```

Comment:
Not blocking (and I know I wrote this in the first place 😄 ) but I think it might be better to have a test like this for each type individually in the `table_options` package. That way, when a new type is added, we can copy and modify the tests there and make sure that the underscore parsing functionality is preserved for new types as well.

Or maybe, even better, would be a single test here that marshals the entire `TableOptions` object, does the inverse operation of `changeCaseForObject`, and checks that when you load that JSON the object is the same? Then we don't have to remember to add new tests for new types and it can be validated for all the fields.

Again, not blocking, I'm happy to follow up and add a test like this after this is merged if you agree it's a good idea.

Reply:
Added this test in https://github.com/cloudquery/cloudquery/pull/10564/commits/96739188c1e7b530c097687c27e2b4ce3fd5e513 and fixed an edge case caught by it in https://github.com/cloudquery/cloudquery/pull/10564/commits/a614f7759b4911694113f7728f6ab52906063814